### PR TITLE
[libc] move bcmp, bzero, bcopy, index, rindex, strcasecmp, strncasecmp to strings.h

### LINKED ISSUE #118875: [libc] split strings.h from string.h
bcmp, bzero, bcopy, index, rindex, strcasecmp, and strncasecmp are technically supposed to be from strings.h, not string.h.

If we move these into a src/strings/ dir, then docgen can find that we have them implemented.

### PR DESCRIPTION
**Pull request by nickdesaulniers**
docgen relies on the convention that we have a file foo.cpp in
libc/src/\<header\>/. Because the above functions weren't in libc/src/strings/
but rather libc/src/string/, docgen could not find that we had implemented
these.

Rather than add special carve outs to docgen, let's fix up our sources for
these 7 functions to stick with the existing conventions the rest of the
codebase follows.

Link: #118860
Fixes: #118875

### CODE REVIEW

File: libc/config/baremetal/arm/entrypoints.txt
Comment (nickdesaulniers) [inline libc/config/baremetal/arm/entrypoints.txt:79]: @petrhosek so if your goal is to avoid POSIX in baremetal, then we can make llvm-libc slightly smaller for baremetal by excluding bcopy, index, rindex, strcasecmp, and strncasecmp which are POSIX and not ISO C.  LLVM may emit libcalls to bcmp (IMO) and bzero (I suspect) (unsure about bcopy, @gchatelet may know; otherwise I know where to look in llvm but am lazy).

File: libc/test/src/string/memory_utils/op_tests.cpp
Comment (nickdesaulniers) [inline libc/test/src/string/memory_utils/op_tests.cpp:249]: @gchatelet thoughts on this question?

**Review by michaelrj-google — COMMENTED** [review]
This looks like a safe NFC to me, but I'd definitely prefer to wait for Guillame to approve.

File: libc/config/baremetal/arm/entrypoints.txt
Comment (gchatelet) [reply libc/config/baremetal/arm/entrypoints.txt:79]: Clang needs to know that these functions [are present](https://github.com/llvm/llvm-project/blob/0f7b3a9407d20e6a4d33ea623e05cf2a3f65eabd/llvm/lib/Analysis/TargetLibraryInfo.cpp#L111-L119). And they are when using glibc or musl libc in the target triple.

File: libc/src/strings/CMakeLists.txt
Comment (gchatelet) [inline libc/src/strings/CMakeLists.txt:6]: should we also add `libc.include.strings`?
Here and below.

File: libc/test/src/string/memory_utils/op_tests.cpp
Comment (gchatelet) [reply libc/test/src/string/memory_utils/op_tests.cpp:249]: It makes sense for unit tests and fuzz tests to move to a `strings` folder but `op_tests` is about testing the low level constructs and I think it's fine to keep them here.

**Review by gchatelet — COMMENTED** [review]
I missed the direct mention, please make sure to ping me directly if you need a review from me.

File: libc/src/strings/CMakeLists.txt
Comment (nickdesaulniers) [reply libc/src/strings/CMakeLists.txt]: Should I just remove having a dependency on our includes? Other functions like `bcopy`, `index`, etc, below don't have such dependencies.  `bcmp` and `bzero` don't explicitly include anything from `libc/include/`.

30655b4a4336

File: libc/test/src/string/memory_utils/op_tests.cpp
Comment (nickdesaulniers) [reply libc/test/src/string/memory_utils/op_tests.cpp]: 24126c4d6da2

File: libc/src/strings/CMakeLists.txt
Comment (gchatelet) [reply libc/src/strings/CMakeLists.txt]: Sure let's remove them then.

**Review by gchatelet — APPROVED** [review]

**Pull request merged**
